src/service/mock-service-proxy/MockServiceExecution.ts

```diff
@@ -94,23 +105,48 @@ export class MockServiceExecution {
                         status: 500,
                         responseBody: { message: errorMessage }
                     }
-                }                
+                }
+
+                if (options && options.headers) {
+                    Object.keys(options.headers).forEach((header) => {
+                        const lowerCaseHeader = header.toLocaleLowerCase();
+
+                        if (lowerCaseHeader === 'accept') {
+                            acceptHeaderOverridden = true;
+                        }
+
+                        if (lowerCaseHeader === 'content-type') {
+                            contentTypeHeaderOverridden = true;
+                        }
+                    });
+                }
+
+                if (!contentTypeHeaderOverridden) {
+                    defaultHeaders['content-type'] = 'application/json';
+                }
+
+                if (!acceptHeaderOverridden) {
+                    defaultHeaders['accept'] = 'application/json';
+                }
+
+                let headers: IHttpHeaders = { ...defaultHeaders, ...this.globalHeaders, ...headersFromOptions };
 
                 this.loggedCalls.push({
                     urlMatches,
                     response,
-                    requestBody
+                    requestBody,
+                    headers
```

Comment:
It might be worth naming this `requestHeaders` to make match `requestBody` and make it clear that it contains only request headers and not response headers.